### ISSUE TITLE
BitCode cannot be enabled in the project with PayPal integration

### ISSUE THREAD

**Issue by UrK** [OceanLabs/iOS-Print-SDK#247]
It seems, PayPal iOS SDK has enabled BitCode integration since 2.12.2:
https://github.com/paypal/PayPal-iOS-SDK/issues/284

Though , when adding PayPal integration of Kite with BitCode the following error is shown by the linker:

    ld: -undefined and -bitcode_bundle (Xcode setting ENABLE_BITCODE=YES) cannot be used together clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by kkarayannis** [comment]
Weird but this issue doesn't happen for me in a new test project. Here is the complete podile I'm using:
```ruby
use_frameworks!

pod 'Kite-Print-SDK'
pod 'Kite-Print-SDK/ApplePay'
pod 'Kite-Print-SDK/Facebook'
pod 'Kite-Print-SDK/Instagram'
pod 'Kite-Print-SDK/ImageProviders'
pod 'Kite-Print-SDK/PayPal'

pre_install do |installer|
    def installer.verify_no_static_framework_transitive_dependencies; end
end
```

It installs and builds with no error on Xcode 7.3 and CocoaPods 0.39.0

BitCode is enabled for the main project.

Can you share anything that can help track down the issue?

**Comment by laud** [comment]
Hi Konstadinos,

Please check out this fresh project (https://github.com/laud/TestPods). I created it with Xcode 7.3 and CocoaPods 0.39.0. 

I get the following error on build:
```
Ld /Users/dlau/Library/Developer/Xcode/DerivedData/TestPods-dlecsynfymggfqciyvbavbquznat/Build/Products/Debug-iphoneos/Kite_Print_SDK.framework/Kite_Print_SDK normal arm64
    cd /Users/dlau/Development/personal-projects/TestPods/Pods
    export IPHONEOS_DEPLOYMENT_TARGET=8.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch arm64 -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS9.3.sdk -L/Users/dlau/Library/Developer/Xcode/DerivedData/TestPods-dlecsynfymggfqciyvbavbquznat/Build/Products/Debug-iphoneos -F/Users/dlau/Library/Developer/Xcode/DerivedData/TestPods-dlecsynfymggfqciyvbavbquznat/Build/Products/Debug-iphoneos -F/Users/dlau/Development/personal-projects/TestPods/Pods/PayPal-iOS-SDK -filelist /Users/dlau/Library/Developer/Xcode/DerivedData/TestPods-dlecsynfymggfqciyvbavbquznat/Build/Intermediates/Pods.build/Debug-iphoneos/Kite-Print-SDK.build/Objects-normal/arm64/Kite_Print_SDK.LinkFileList -install_name @rpath/Kite_Print_SDK.framework/Kite_Print_SDK -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -miphoneos-version-min=8.0 -dead_strip -Xlinker -no_deduplicate -fembed-bitcode-marker -undefined dynamic_lookup -fobjc-arc -fobjc-link-runtime -framework ApplePayStubs -framework CSStickyHeaderFlowLayout -framework DACircularProgress -framework Foundation -framework ios_qr_encoder -framework KITAssetsPickerController -framework MPFlipViewController -framework SDWebImage -framework SkyLab -framework Stripe -framework SVProgressHUD -framework TSMarkdownParser -framework UICKeyChainStore -framework UIColor_HexString -single_module -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/dlau/Library/Developer/Xcode/DerivedData/TestPods-dlecsynfymggfqciyvbavbquznat/Build/Intermediates/Pods.build/Debug-iphoneos/Kite-Print-SDK.build/Objects-normal/arm64/Kite_Print_SDK_dependency_info.dat -o /Users/dlau/Library/Developer/Xcode/DerivedData/TestPods-dlecsynfymggfqciyvbavbquznat/Build/Products/Debug-iphoneos/Kite_Print_SDK.framework/Kite_Print_SDK

ld: -undefined and -bitcode_bundle (Xcode setting ENABLE_BITCODE=YES) cannot be used together
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```


If I disable ENABLE_BITCODE on both the app target and Kite-Print-SDK target, I get the following runtime error:

```
dyld: Symbol not found: _PayPalEnvironmentProduction
  Referenced from: /private/var/containers/Bundle/Application/18C7FDA9-75D5-4D70-A5D4-8A1C3C2431AD/TestPods.app/Frameworks/Kite_Print_SDK.framework/Kite_Print_SDK
  Expected in: flat namespace
 in /private/var/containers/Bundle/Application/18C7FDA9-75D5-4D70-A5D4-8A1C3C2431AD/TestPods.app/Frameworks/Kite_Print_SDK.framework/Kite_Print_SDK
```

**Comment by NickBols** [comment]
I have issues with Paypal SDK as well. Maybe related to this one?

**Cocoapods install works with podfile like so:
**
platform :ios, "8.0"
use_frameworks!

target 'myProject' do

pod 'Kite-Print-SDK'
pod 'Kite-Print-SDK/ApplePay'
pod 'Kite-Print-SDK/Facebook'
pod 'Kite-Print-SDK/Instagram'
pod 'Kite-Print-SDK/ImageProviders'
pod 'Kite-Print-SDK/PayPal'

pre_install do |installer|
def installer.verify_no_static_framework_transitive_dependencies; end
end
end

**But after install: 
**
7 Errors: 
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_PayPalPaymentViewController", referenced from:
      objc-class-ref in OLPaymentViewController.o
  "_OBJC_CLASS_$_PayPalConfiguration", referenced from:
      objc-class-ref in OLPaymentViewController.o
  "_PayPalEnvironmentSandbox", referenced from:
      +[OLKitePrintSDK paypalEnvironment] in OLKitePrintSDK.o
  "_OBJC_CLASS_$_PayPalPayment", referenced from:
      objc-class-ref in OLPaymentViewController.o
  "_PayPalEnvironmentProduction", referenced from:
      +[OLKitePrintSDK paypalEnvironment] in OLKitePrintSDK.o
  "_OBJC_CLASS_$_PayPalMobile", referenced from:
      objc-class-ref in OLPaymentViewController.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

The Other linker flags in target -> build settings were weirdly typed for example -l"PayPalMobile" and -l"c++ so I tried changing those to the normal ones you usually add for the paypal SDK but still the same errors..

Any ideas?

**Comment by kkarayannis** [comment]
As far as I can tell, there is no way to integrate PayPal when using `use_frameworks!`
The problem seems to be that a dynamic framework (Kite-Print-SDK) cannot be linked against a static library (PayPal). People have asked for a dynamic framework from PayPal [here](https://github.com/paypal/PayPal-iOS-SDK/issues/414).

Alternatively you can try integrating using git submodules.

**Comment by NickBols** [comment]
Ahh i see thanks.. Hope that comes soon but just gonna leave it out for now then too much trouble and things that can go wrong.. CC and ApplePay will be fine!

**Comment by nlminh** [comment]
I think with PayPal-iOS-SDK/Core version 2.14.2 and iOS 8.0+ target people can use dynamic framework : [hier](https://github.com/paypal/PayPal-iOS-SDK/commit/641d5f47b45ff91acdb9d663b6d4088af19750b9)

```
- `PayPal-iOS-SDK/Core` required by `Podfile`
- `PayPal-iOS-SDK/Core (= 2.14.2)` required by `Podfile.lock`
- `PayPal-iOS-SDK/Core (~> 2.12.6)` required by `Kite-Print-SDK/PayPal (5.1.1)`
- `Kite-Print-SDK/PayPal` required by `Podfile`
```

Can you update this?

**Comment by kkarayannis** [comment]
This should now be solved with v6.0.0 of the SDK. Please note that PayPal should be installed directly so use:
```
pod 'PayPal-iOS-SDK/Core', '~> 2.12.5'
```

and not:
```
pod 'Kite-Print-SDK/PayPal'
```
